### PR TITLE
Tooling for issuing JWT.

### PR DESCRIPTION
**Pull request by lightsofapollo**
Adds tooling for issuing JWT tokens.

 - The private key is used to sign the JWT token but is never stored (see #102)
 - We save the public key, key id (used to identify the JWT), issue date in the database.

Authorization will look like this:

 - We check (routes which require auth) for a bearer token (JWT).
 - If the bearer token is missing HTTP 401.
 - We extract the "keyId" (kid) from the JWT.
 - We fetch the key from the database with the matching "keyId".
 - We verify the JWT against the public key we have stored.

The JWT passed acts much like a cookie or session token it tells us who the token belongs to and provides us a signature to verify that the JWT was indeed created by the server.

I chose ES256 carefully after looking at the alternatives... https://github.com/mozilla-iot/gateway/issues/102 contains some more rational.

NOTE: I plan on leaving expiry of tokens to a future PR.

### CODE REVIEW

**Comment by lightsofapollo** [comment]
A side note... If you are using ES2015 features "use strict" is implied (basically you don't need to use "use strict" anymore...).

File: ec-crypto.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+const asn1 = require('asn1.js');
```
Comment (benfrancis) [inline ec-crypto.js]: Would you mind adding a header to this file, just for consistency:

```
/**
 * [an appropriate name].
 *
 * [a description of what this file does].
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
```

File: models/jsonwebtoken.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+const assert = require('assert');
```
Comment (benfrancis) [inline models/jsonwebtoken.js]: Again, explanatory file header would be nice,

**Comment by lightsofapollo** [comment]
I snuck babel-eslint in here at the end... It does slow down linting but with my "fastlint" pr it goes from 1.2s to 1.7s (acceptable).

File: .eslintrc.js [hunk 3]
```diff
@@ -4,8 +4,10 @@ module.exports = {
     'es6': true,
     'node': true,
     'browser': true,
+    'mocha': true,
   },
   'extends': 'eslint:recommended',
+  'parser': 'babel-eslint',
```
Comment (hobinjk) [inline .eslintrc.js]: `babel-eslint` should be added to package.json

**Pull request merged**